spec/components/card_component_spec.rb

```diff
@@ -10,58 +10,102 @@
     }.with_indifferent_access
   end
 
-  subject! { render_inline(CardComponent.new(card: card)) }
+  let(:card) { base }
+
+  subject do
+    render_inline CardComponent.new(card: card)
+    page
+  end
 
   specify "renders a card" do
-    expect(page).to have_css(".card")
+    is_expected.to have_css(".card")
   end
 
   specify "includes a link wrapping the story image" do
-    expect(page).to have_link(href: card[:link]) do |anchor|
+    is_expected.to have_link(href: card[:link]) do |anchor|
       expect(anchor).to have_css(%(img[src='#{card[:image]}']))
     end
   end
 
   specify "includes the snippet" do
-    expect(page).to have_content(card[:snippet])
+    is_expected.to have_content(card[:snippet])
   end
 
   specify "includes the linktext in the link" do
-    expect(page).to have_link(card[:link_text], href: card[:link], class: "git-link")
+    is_expected.to have_link(card[:link_text], href: card[:link], class: "git-link")
   end
 
   specify "no play icon is visible" do
-    expect(page).not_to have_css(".fas.fa-play")
+    is_expected.not_to have_css(".fas.fa-play")
+  end
+
+  specify "header does not show by default" do
+    is_expected.not_to have_css(".card header")
+  end
+
+  specify "border shows by default" do
+    is_expected.not_to have_css(".card.card--no-border")
+  end
+
+  context "with border removed" do
+    let(:card) { base.merge(border: false) }
+
+    specify "border is removed" do
+      is_expected.to have_css(".card.card--no-border")
+    end
+  end
+
+  context "with header set" do
+    let(:card) { base.merge(header: "Further info") }
+
+    specify "header is shown" do
+      is_expected.to have_css(".card header")
```

Comment:
I think it makes sense to check the text matches here too

Reply:
Good call